### PR TITLE
Made available MicaConfig as JSON to search page.

### PR DESCRIPTION
**Pull request by kazoompa**

### CODE REVIEW

File: mica-webapp/src/main/resources/templates/search.ftl [hunk 1]
```diff
@@ -272,6 +272,9 @@
 </div>
 <!-- ./wrapper -->
 
+<!-- MicaConfig in JSON Format -->
+<input type="hidden" id="mica-config" value='${configJson}'></input>
```
Comment (ymarcon) [inline mica-webapp/src/main/resources/templates/search.ftl]: if you remove the quotes you will have a JSON object directly: no need to parse it afterward.

File: mica-webapp/src/main/java/org/obiba/mica/web/controller/SearchController.java [hunk 2]
```diff
@@ -1,15 +1,56 @@
...
+
 @Controller
 public class SearchController {
 
+  private MicaConfigService micaConfigService;
+
+  private String cachedMicaConfigJson = null;
+
+  @Inject
+  public SearchController(MicaConfigService micaConfigService) {
```
Comment (ymarcon) [inline mica-webapp/src/main/java/org/obiba/mica/web/controller/SearchController.java]: Why not using the MicaConfigInterceptor where the MicaConfig object is already extracted and cached?

File: mica-webapp/src/main/resources/templates/search.ftl (hunk 1 above)
Comment (ymarcon) [reply mica-webapp/src/main/resources/templates/search.ftl]: And why not making a const directly? What's the use of this input element?

File: mica-webapp/src/main/java/org/obiba/mica/web/controller/SearchController.java (hunk 2 above)
Comment (kazoompa) [reply mica-webapp/src/main/java/org/obiba/mica/web/controller/SearchController.java]: I updated my branch and removed the cache.

**Pull request closed without merging**